runners/flink/src/main/java/org/apache/beam/runners/flink/translation/wrappers/streaming/DoFnOperator.java

```diff
@@ -455,6 +461,18 @@ public void open() throws Exception {
     if (!options.getDisableMetrics()) {
       flinkMetricContainer = new FlinkMetricContainer(getRuntimeContext());
       doFnRunner = new DoFnRunnerWithMetricsUpdate<>(stepName, doFnRunner, flinkMetricContainer);
+      String checkpointMetricNamespace =
```

Comment:
I thought about that. The checkpointing metrics are kept at the JobManager. They are not accessible by the operator. So in order not to change the logic of the load tests which relies on the metrics accumulated during the job, returned in the PipelineResult, I opted to go this route. 

I think we may have to change the Beam service to allow querying job metrics during job runtime. At the moment, we can only retrieve them at the end of the job.

Reply:
It would be good to expose/utilize the Flink metrics as part of the load test suite. This could, for example, be accomplished with a custom Flink metric reporter. It can be addressed as follow-up though.

Reply:
Yes, I agree that we want this to be more generic. I didn't want to mention the Flink metric reporter but I think that would introduce another level of complexity. Not sure if that is the right tool for the job. I'd rather query metrics through the job server and the rest api.